File: tests/fixtures/edgecase_repo/cpp/module3.cpp
#include <string>
#include "module2.hpp"

namespace gen {

class Base3 {
public:
    std::string name;
};

class Service3 : public Base3 {
public:
    bool handle(const std::string &payload) {
        this->validate(payload);
        return true;
    }

    void validate(const std::string &payload) {}
};

int helper_3(int value) { return value; }

}  // namespace gen

File: tests/fixtures/edgecase_repo/php/module3.php
<?php
namespace Example\Gen;

use Example\Gen\Base3;
use Example\Gen\Handler3 as H3;

class Base3
{
    protected $name;
}

interface Handler3
{
    public function handle($payload);
}

class Service3 extends Base3 implements Handler3
{
    public function handle($payload): bool
    {
        $this->validate($payload);
        return true;
    }

    private function validate($payload): void
    {
    }
}

function helper_3($value) {
    return $value;
}
